src/utils/decorators/input-validation/input-validation.js

```diff
@@ -225,20 +229,13 @@ const InputValidation = (ComposedComponent) => {
       }
     }
 
-    /**
-     * A lifecycle method for when the component is added to the page.
-     *
-     * @method componentWillMount
-     * @return {void}
-     */
-    componentWillMount() {
-      // call the components super method if it exists
-      /* istanbul ignore else */
-      if (super.componentWillMount) { super.componentWillMount(); }
-
-      if (this.context.form && (this._validations() || this.props.warnings || this.props.info)) {
-        // attach the input to the form so the form can track what it needs to validate on submit
-        this.context.form.attachToForm(this);
+    initValidationInput = (context) => {
+      if (!this.newContextIntegration && context) {
+        this.newContextIntegration = context;
+        if (this._validations() || this.props.warnings || this.props.info) {
+          // attach the input to the form so the form can track what it needs to validate on submit
+          context.addInput(this._guid, this.validateFromForm);
+        }
```

Comment:
that block could be extracted into a separate method to avoid nesting conditionals